src/test/java/org/opentripplanner/middleware/ApiUserFlowTest.java

```diff
@@ -0,0 +1,169 @@
+package org.opentripplanner.middleware;
+
+import com.auth0.exception.Auth0Exception;
+import com.auth0.json.mgmt.users.User;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.middleware.models.AdminUser;
+import org.opentripplanner.middleware.models.ApiUser;
+import org.opentripplanner.middleware.models.MonitoredTrip;
+import org.opentripplanner.middleware.models.OtpUser;
+import org.opentripplanner.middleware.models.TripRequest;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.persistence.PersistenceUtil;
+import org.opentripplanner.middleware.utils.HttpUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+
+import java.io.IOException;
+import java.net.http.HttpResponse;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static org.opentripplanner.middleware.OtpMiddlewareMain.getBooleanEnvVar;
+import static org.opentripplanner.middleware.TestUtils.mockAuthenticatedPost;
+import static org.opentripplanner.middleware.TestUtils.mockAuthenticatedRequest;
+import static org.opentripplanner.middleware.auth.Auth0Connection.authDisabled;
+import static org.opentripplanner.middleware.auth.Auth0Users.createAuth0UserForEmail;
+import static org.opentripplanner.middleware.controllers.api.ApiUserController.DEFAULT_USAGE_PLAN_ID;
+
+/**
+ * Tests to simulate API user flow. The following config parameters must be set in configurations/default/env.yml for
+ * these end-to-end tests to run: - AUTH0_DOMAIN set to a valid Auth0 domain - AUTH0_API_CLIENT set to a valid Auth0
+ * application client id - AUTH0_API_SECRET set to a valid Auth0 application client secret - DEFAULT_USAGE_PLAN_ID set
+ * to a valid usage plan id (AWS requires this to create an api key). - An AWS_PROFILE is required, or AWS access has
+ * been configured for your operating environment e.g. C:\Users\<username>\.aws\credentials in Windows or Mac OS
+ * equivalent.
+ *
+ * The following environment variable must be set for these tests to run: - RUN_E2E=true
+ *
+ * Auth0 must be correctly configured as described here: https://auth0.com/docs/flows/call-your-api-using-resource-owner-password-flow
+ */
+public class ApiUserFlowTest {
+    private static ApiUser apiUser;
+    private static OtpUser otpUser;
+
+    /**
+     * Create an {@link ApiUser} and an {@link AdminUser} prior to unit tests
+     */
+    @BeforeAll
+    public static void setUp() throws IOException {
+        OtpMiddlewareTest.setUp();
```

Comment:
Needs a `assumeTrue(getBooleanEnvVar("RUN_E2E"));` before the `OtpMiddlewareTest.setUp();` so that this only runs during E2E testing.

Reply:
Updated.